### ISSUE TITLE
[Qustion] Overfitting problem

### ISSUE THREAD

**Issue by NeighborhoodCoding** [lidq92/VSFA#38]
Hi, I'm trying to enhance validation mae error, recently I discovered that konvid 1k dataset has overfitting problem. (I'm also trying to train CNN layer also...)

In my specific case, the train mae is reduced to 0.01/1.0 but validation mae is 0.1/1.0 when the epoch is increased.

I think that the small number of train data(total 1200 ea) is one of the problem.

I'm thinking data augmentation like this : https://github.com/okankop/vidaug

Because konvid-1k dataset has MOS label, I think vertical or horizontal flip may good for reduce val mae..? (Not add noise level)

This may good way...? I'm trying to add this in the your provided code and will check this way may reduce val mae. Thanks you!!

**Comment by lidq92** [comment]
@SungmanHong Small rotation or other transforms would be OK if they would not change the video quality (e.g., small enough that below JND). If you still encounter the overfitting problem, for a better tradeoff, you may "relax" your data augmentation to be the ones that would not significantly change the video quality (e.g., just near the JND). Otherwise, you have to use other strategies to avoid this problem, e.g., regularizing the network or construct a larger training set.

You may want to use the larger datasets constructed recently, e.g., [KonVid-150k: A Dataset for No-Reference Video Quality Assessment of Videos in-the-Wild](http://arxiv.org/abs/1912.07966v2) IEEE Access 2021 & [Patch-VQ: 'Patching Up' the Video Quality Problem](https://arxiv.org/abs/2011.13544) CVPR 2021. 
VQA is not the main focus of my current research, and I haven't tries these new datasets. 
Hope you would find this information helpful.

Good luck,
Dingquan

**Comment by NeighborhoodCoding** [comment]
Thank you, I think konvid150k is not available to download...  I'll recheck ovrfitting problem of konvid1k datasets.  First I will load well trained VSFA without CNN training model and set learning rate or require_grad to only CNN layers and train only CNN layers and check validation performance is better or  not. If I finish it, I will share the results. Thank you!

**Comment by NeighborhoodCoding** [comment]
Thanks, I add 3D CNN layers in LSTM class. (3D : Frames X Width X Height)
Before 3 days ago, I added 2D CNN layers and for loop( range 1 to  len(frames)).

After I changed 2D CNN to 3D CNN, the performance is much better and I think overfitting problem is resolved a lot.
However, the performance is generally not much good, it seems, maybe a little(1~2%) better or worse...?
My train is now in 77 epoch and train mae is 0.0472/1.0 and validation mae is  0.052/1.0
validation mae is more higher than train mae, and the overfitting problem seems not very high now in KoNViD 1k datasets. 

I'm not sure how to more improve mae error, but now I'm trying to add new features like natural scene statistics.... 

Although my opinion is just my opinion, I don't think the overfitting problem is very serious for KoNViD 1k datasets now(CNN seems can be trained on KoNViD 1k datasets now), so I'm going to close this issue. Although data augmentation may help performance. I'm very newbie in pytorch;; Thank you for your help.

**Comment by FranzHahn** [comment]
Hi guys, creator of KonVid-150k here. We recently made the video dataset public at http://database.mmsp-kn.de/konvid-150k-vqa-database.html.

Please let me know via e-mail (hahn [dot] franz [at] gmail [dot] com) if you have any questions!
